### PR TITLE
`azurerm_virtual_network_gateway_connection` Add new feature to set `custom_bgp_addresses`

### PR DESCRIPTION
**Pull request by HappyTobi**
Hi all,

I just added the new Azure feature to set a "custom_bgp_address" to the virtual network gateway connection.

The configuration looks like:
```
resource "azurerm_virtual_network_gateway_connection" "TestTunnelInstance" {
  name                = "TestTunnelInstance"
  resource_group_name = azurerm_resource_group.TestRG1.name
  location            = azurerm_resource_group.TestRG1.location

  type                       = "IPsec"
  virtual_network_gateway_id = azurerm_virtual_network_gateway.VNet1GW.id
  local_network_gateway_id   = azurerm_local_network_gateway.TestGateway.id
  shared_key                 = "SECRET"

  enable_bgp = true

  custom_bgp_addresses {
    primary_address   = "169.254.22.2"
    secondary_address = "169.254.22.6"
  }
}

```

Azure Documentation about that topic:
https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-howto-aws-bgp?msclkid=11de8e64c6f411eca4c7b7511dbc44d7

I also had to update the hole Networking api vom `2021-05-01` to `2021-08-01`

The test runs fine when I locally updated the vendor dependencies.

I don't know if I should check in the updated dependencies, if that is required please let me know.

Regards,
Tobi

### CODE REVIEW

**Comment by HappyTobi** [comment]
Hi @katbyte,
is it possible that  someone of you can review the pr?

Regards
Tobi

File: internal/services/network/virtual_network_gateway_connection_resource.go
Comment (katbyte) [inline internal/services/network/virtual_network_gateway_connection_resource.go:200]: as address is in the containing block:
```suggestion
						"primary": {
							Type:         pluginsdk.TypeString,
							Required:     true,
							ValidateFunc: validation.Any(validation.IsIPv4Address),
						},
						"secondary": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the PR @HappyTobi - aside from 1 comment on schema names this LGTM! once thats fixed up this should be good to merge

File: internal/services/network/virtual_network_gateway_connection_resource.go
Comment (HappyTobi) [reply internal/services/network/virtual_network_gateway_connection_resource.go:200]: I changed all the properties like the suggestion

**Review by katbyte — APPROVED** [review]
Thanks @HappyTobi - LGTM 🍰

**Comment by katbyte** [comment]
network API upgrade had a breaking change for application gateway: #16793

**Pull request merged**
